utils/helpers/userHelper.ts

```diff
@@ -0,0 +1,53 @@
+import {UserProfile} from "@auth0/nextjs-auth0/src/frontend/use-user";
+import {db} from "../../firebase";
+
+export const getUserFromFirebase = async (user: UserProfile) => {
+    try {
+        //creates user if not created.
+        db.collection("auth0").where('auth0', '==', user.sub).get().then((data) => {
+            if (!data.docs.length) {
+                // to create Auth0 of the new user.
+                db.collection('auth0').doc().set({
+                    auth0: user.sub,
+                }).then(() => {
+                    console.log('auth0 created')
+                })
+                // to create new user.
+                db.collection('users').doc().set({
```

Comment:
This is where I'd change. I wouldn't have
`
uid: user.sub
`
but I would use the doc id of the auth0 collection as doc id for the users collection.
To specify the doc id you can do this:
`
db.collection("cities").doc("LA").set({
    name: "Los Angeles",
    state: "CA",
    country: "USA"
})
`
basically pass the id to "doc()".

Also, we should use firebase v9 syntax if possible.

Reply:
To do the above and using Firebase v9+ syntax we want to:

1. create an entry in the auth0 collection
2. get the doc id for the newly created mapping
3. create a user and profile doc with same doc id

`
// 1 - Add auth0 mapping  
const docRef = await addDoc(collection(db, 'auth0'), {
      auth0: user.sub,
})

// 2, 3 - Add user with doc id above
const data = {
  email: '',
  expirationTime: '',
  ...
}
await setDoc(doc(db, "users", docRef.id), data);

// Similar for profile
`
More info at Firebase [documentation](https://firebase.google.com/docs/firestore/manage-data/add-data#web-version-9_6)